src/onnx/parse_pooling.cpp

```diff
@@ -29,16 +29,15 @@ struct parse_pooling : op_parser<parse_pooling>
                           onnx_parser::node_info info,
                           std::vector<instruction_ref> args) const
     {
-        std::vector<std::string> valid_modes = {"max", "average", "lpnorm"};
-        std::string mode                     = opd.op_name;
-        if(std::find(valid_modes.begin(), valid_modes.end(), mode) == valid_modes.end())
-        {
-            MIGRAPHX_THROW("onnx pooling mode must be [\"max\", \"average\", \"lpnorm\"]");
-        }
         const std::unordered_map<std::string, op::pooling_mode> mode_map = {
             {"max", op::pooling_mode::max},
             {"average", op::pooling_mode::average},
             {"lpnorm", op::pooling_mode::lpnorm}};
+        std::string mode = opd.op_name;
+        if(mode_map.find(mode) == mode_map.end())
```

Comment:
I think you can write `if (not contains(mode_map, mode)`.

Reply:
That looks to be a C++20 addition: https://en.cppreference.com/w/cpp/container/unordered_map/contains. I think we're using C++17?

Reply:
I am referring to the `contains` we have function here:

https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/develop/src/include/migraphx/ranges.hpp#L95